NVorbis/Contracts/IStreamDecoder.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+
+namespace NVorbis.Contracts
+{
+    /// <summary>
+    /// Describes a stream decoder instance for Vorbis data.
+    /// </summary>
+    public interface IStreamDecoder : IDisposable
+    {
+        /// <summary>
+        /// Gets the number of channels in the stream.
+        /// </summary>
+        int Channels { get; }
+
+        /// <summary>
+        /// Gets the sample rate of the stream.
+        /// </summary>
+        int SampleRate { get; }
+
+        /// <summary>
+        /// Gets the upper bitrate limit for the stream, if specified.
+        /// </summary>
+        int UpperBitrate { get; }
+
+        /// <summary>
+        /// Gets the nominal bitrate of the stream, if specified.  May be calculated from <see cref="LowerBitrate"/> and <see cref="UpperBitrate"/>.
+        /// </summary>
+        int NominalBitrate { get; }
+
+        /// <summary>
+        /// Gets the lower bitrate limit for the stream, if specified.
+        /// </summary>
+        int LowerBitrate { get; }
+
+        /// <summary>
+        /// Gets the tag data from the stream's header.
+        /// </summary>
+        ITagData Tags { get; }
+
+        /// <summary>
+        /// Gets the total duration of the decoded stream.
+        /// </summary>
+        TimeSpan TotalTime { get; }
+
+        /// <summary>
+        /// Gets the total number of samples in the decoded stream.
+        /// </summary>
+        long TotalSamples { get; }
+
+        /// <summary>
+        /// Gets or sets the current time position of the stream.
+        /// </summary>
+        TimeSpan TimePosition { get; set; }
+
+        /// <summary>
+        /// Gets or sets the current sample position of the stream.
+        /// </summary>
+        long SamplePosition { get; set; }
+
+        /// <summary>
+        /// Gets or sets whether to clip samples returned by <see cref="Read(float[], int, int)"/>.
+        /// </summary>
+        bool ClipSamples { get; set; }
+
+        /// <summary>
+        /// Gets whether <see cref="Read(float[], int, int)"/> has returned any clipped samples.
+        /// </summary>
+        bool HasClipped { get; }
+
+        /// <summary>
+        /// Gets whether the decoder has reached the end of the stream.
+        /// </summary>
+        bool IsEndOfStream { get; }
+
+        /// <summary>
+        /// Gets the <see cref="IStreamStats"/> instance for this stream.
+        /// </summary>
+        IStreamStats Stats { get; }
+
+        /// <summary>
+        /// Seeks the stream to the specified time.
+        /// </summary>
+        /// <param name="timePosition">The time to seek to.</param>
+        void SeekTo(TimeSpan timePosition);
+
+        /// <summary>
+        /// Seeks the stream to the specified sample.
+        /// </summary>
+        /// <param name="samplePosition">The sample position to seek to.</param>
+        void SeekTo(long samplePosition);
```

Comment:
Is there a difference between these two methods and just setting through properties? Do the properties not seek the underlying stream? Maybe add [SeekOrigin ](https://docs.microsoft.com/en-us/dotnet/api/system.io.seekorigin?view=netframework-4.8)for consistency with .Net?

Reply:
Hmmm...  While that would technically be a breaking change, I think it's a good idea.  That also makes having the separate methods (which are currently identical to the property setters) "make sense".  I'll see about adding the parameter.